File: soundnotification/src/main/java/org/bug/soundnotification/ServiceBootstrap.java
package org.bug.soundnotification;

import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;
import android.util.Log;

public class ServiceBootstrap extends BroadcastReceiver {

	@Override
	public void onReceive(Context context, Intent intent) {
		Log.d(getClass().getName(), "Received " + context + " " + intent);
		try {
			if (intent != null && Intent.ACTION_BOOT_COMPLETED.equals(intent.getAction()) &&
					!NotificationService.isStarted()) {
				Intent serviceIntent = new Intent(context, NotificationService.class);
				context.startService(serviceIntent);
				NotificationService.setStarted(true);
			}
		} catch (Exception e) {
			Log.e(getClass().getName(), "Receive error", e);
		}
	}
}

File: soundnotification/src/main/java/org/bug/soundnotification/ServiceBootstrap.kt
package org.bug.soundnotification

import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent
import android.util.Log
import androidx.preference.PreferenceManager

class ServiceBootstrap : BroadcastReceiver() {
    override fun onReceive(ctx: Context, intent: Intent) {
        Log.d(javaClass.name, "Received $ctx $intent")
        val enabled = PreferenceManager.getDefaultSharedPreferences(ctx).getBoolean("enable_checkbox", false)
        if (Intent.ACTION_BOOT_COMPLETED == intent.action && enabled) {
            startService(ctx)
        }
    }

    companion object {
        fun startService(ctx: Context) {
            val serviceIntent = Intent(ctx, NotificationService::class.java)
            ctx.startService(serviceIntent)
        }
    }
}